### ISSUE TITLE
Proposal: Ability to set params that can be used within the translation interpolation. 

### ISSUE THREAD

**Issue by hworld** [Polyconseil/vue-gettext#17]
I like to set up translation strings that are agnostic to how my app state is structured. This way if I change variable names within my app, it won't break my translation strings.

Here's an example of what I would like to do:

```
<translate
	:translate-params="{ count: number( listing.gamesCount ), page: number( listing.currentPage ) }"
	:translate-n="listing.gamesCount"
	translate-plural="Page %{ page } of %{ count } games."
	>
	%{ count } game.
</translate>
```

You can see above that it also allows you to really easily modify the variables before passing in to the translations. I just use a simple function above that converts a number to display correctly for their locale as an example.

I wrote out a version of this that seems to be working. I haven't tested it extensively yet, but it does watch and update the translation correctly when the params change:

https://github.com/gamejolt/vue-gettext/commit/cabcc9b385aa8e49b233ceca96b89671bf1e0bb7

**Comment by kemar** [comment]
@hworld yes, this looks like a cool enhancement.

If I have time I may work on it.

**Comment by hworld** [comment]
Just going to update this a bit. It's been working really well for me to do the method I said above. I just updated what I have to use the directive as well. Here's the full changes: https://github.com/gamejolt/vue-gettext/commit/ba4e9e01e372a2d3ff7eb8cb67cc2f4c68203aa0

Pretty minimal. For the directive version I'm just using the binding value from the directive. I think this will also make sure that the binding actually updates when the params change, so I think this is actually better than just using the vnode context directly.

Here is how the directive version would look: 
```html
<span
	v-translate="{ count: number( gamesCount ), page: number( currentPage ) }"
	:translate-n="gamesCount"
	translate-plural="Page %{ page } of %{ count } games."
	>
	<b>%{ count }</b> game.
</span>
```

**Comment by kemar** [comment]
Since nobody else is asking for this feature, I'm not going to work on it for now without any further demand.

If you really want it, you can submit a pull request [following the guidelines](https://github.com/Polyconseil/vue-gettext/blob/master/README_DEV.md#pull-request-guidelines).

**Comment by kennyki** [comment]
@kemar coincidently I stumbled upon a problem that can be solved with the solution proposed:

#### Interpolation inside v-for doesn't work

```html
<p>
  <translate v-for='name in names'>Hello %{name}</translate>
  <span v-for='name in names' v-translate>Hello %{name}</span>
</p>
```

I tried but cannot figure out a way to access the local scope (in this case, `name`) in the component or directive.

So I shamelessly adopted the code from @hworld (hope you don't mind it) and made it to work:

```html
<p>
  <translate v-for='name in names' :translate-params='{name: name}'>Hello %{name}</translate>
  <span v-for='name in names' v-translate='{name: name}'>Hello %{name}</span>
</p>
```

Please check out #36

**Comment by kemar** [comment]
Ah, I guess today is @hworld's lucky day :)

Ok thanx @kennyki I'll review your pull request asap.

**Comment by hworld** [comment]
Glad my fork could be of help! I think injecting params into translation texts is a very sane approach to handling dynamic translations. Sorry I didn't submit a PR. I'm currently working on a complete rewrite of our angular codebase into vue and haven't had time to make PRs anywhere. I was just hoping that my forked hack would be of help if someone were able to get to a PR before me.

Cheers!

**Comment by kemar** [comment]
Implemented in [v2.0.20](https://github.com/Polyconseil/vue-gettext/releases/tag/v2.0.20).

**Comment by kennyki** [comment]
@hworld thanks for introducing an interesting translation strategy. I might try it out too.
